[CI] ShrinkIndexIT testShrinkThenSplitWithFailedNode failure

Build scan: https://gradle.com/s/k5loufononht4
Console log:  https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=zulu12,nodes=general-purpose/98/console

Failure:
```
07:53:55   2> REPRODUCE WITH: ./gradlew :server:integTest --tests "org.elasticsearch.action.admin.indices.create.ShrinkIndexIT.testShrinkThenSplitWithFailedNode" -Dtests.seed=7F9B0926CAB77C2 -Dtests.security.manager=true -Dtests.locale=nus -Dtests.timezone=America/Indiana/Knox -Dcompiler.java=12 -Druntime.java=12
07:53:55   2> java.lang.AssertionError: ResizeResponse failed - not acked
07:53:55     Expected: <true>
07:53:55          but: was <false>
07:53:55         at __randomizedtesting.SeedInfo.seed([7F9B0926CAB77C2:DAC471C7C22737B0]:0)
07:53:55         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
07:53:55         at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked(ElasticsearchAssertions.java:112)
07:53:55         at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked(ElasticsearchAssertions.java:100)
07:53:55         at org.elasticsearch.action.admin.indices.create.ShrinkIndexIT.testShrinkThenSplitWithFailedNode(ShrinkIndexIT.java:589)
07:53:55   1> [2019-07-22T23:53:46,590][INFO ][o.e.a.a.i.c.ShrinkIndexIT] [testCreateShrinkIndexToN] before test
07:53:55   1> [2019-07-22T23:53:46,590][INFO ][o.e.a.a.i.c.ShrinkIndexIT] [testCreateShrinkIndexToN] [ShrinkIndexIT#testCreateShrinkIndexToN]: setting up test
07:53:55   1> [2019-07-22T23:53:46,590][INFO ][o.e.t.InternalTestCluster] [testCreateShrinkIndexToN] adding voting config exclusions [node_s2] prior to restart/shutdown
07:53:55   1> [2019-07-22T23:53:46,620][INFO ][o.e.n.Node               ] [testCreateShrinkIndexToN] stopping ...
07:53:55   1> [2019-07-22T23:53:46,621][INFO ][o.e.c.c.Coordinator      ] [node_s2] master node [{node_s0}{0Vn4EYvgS7aBtp3STN22mQ}{0vkYnVenRCm120uqiHNWPg}{127.0.0.1}{127.0.0.1:42453}{dim}] failed, restarting discovery
07:53:55   1> org.elasticsearch.transport.NodeDisconnectedException: [node_s0][127.0.0.1:42453][disconnected] disconnected
07:53:55   1> [2019-07-22T23:53:46,623][INFO ][o.e.c.s.MasterService    ] [node_s0] node-left[{node_s2}{27VqX-qiQaqfZcYaMp-jUQ}{ELWQ7z4oRhazcYurKCwzxg}{127.0.0.1}{127.0.0.1:40475}{dim} disconnected], term: 1, version: 250, reason: removed {{node_s2}{27VqX-qiQaqfZcYaMp-jUQ}{ELWQ7z4oRhazcYurKCwzxg}{127.0.0.1}{127.0.0.1:40475}{dim},}
07:53:55   1> [2019-07-22T23:53:46,623][INFO ][o.e.n.Node               ] [testCreateShrinkIndexToN] stopped
07:53:55   1> [2019-07-22T23:53:46,623][INFO ][o.e.n.Node               ] [testCreateShrinkIndexToN] closing ...
07:53:55   1> [2019-07-22T23:53:46,625][INFO ][o.e.n.Node               ] [testCreateShrinkIndexToN] closed
```

Reproduction with: 

```
./gradlew :server:integTest --tests "org.elasticsearch.action.admin.indices.create.ShrinkIndexIT.testShrinkThenSplitWithFailedNode" \
  -Dtests.seed=7F9B0926CAB77C2 \
  -Dtests.security.manager=true \
  -Dtests.locale=nus \
  -Dtests.timezone=America/Indiana/Knox \
  -Dcompiler.java=12 \
  -Druntime.java=12
```
```
./gradlew :server:integTest --tests "org.elasticsearch.action.admin.indices.create.ShrinkIndexIT.testShrinkThenSplitWithFailedNode" \
  -Dtests.seed=7F9B0926CAB77C2 \
  -Dtests.security.manager=true \
  -Dtests.locale=nus \
  -Dtests.timezone=America/Indiana/Knox \
  -Dcompiler.java=12 \
  -Druntime.java=12
```

This does not reproduce locally. Pinging @original-brownbear because of https://github.com/elastic/elasticsearch/pull/44214

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java
@@ -580,7 +580,9 @@ public class ShrinkIndexIT extends ESIntegTestCase {
             .build()).setResizeType(ResizeType.SHRINK).get());
         ensureGreen();
 
+        final int nodeCount = cluster().size();
         internalCluster().stopRandomNode(InternalTestCluster.nameFilter(shrinkNode));
+        ensureStableCluster(nodeCount -1);
 
         // demonstrate that the index.routing.allocation.initial_recovery setting from the shrink doesn't carry over into the split index,
         // because this would cause the shrink to fail as the initial_recovery node is no longer present.

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java
@@ -582,7 +582,7 @@ public class ShrinkIndexIT extends ESIntegTestCase {
 
         final int nodeCount = cluster().size();
         internalCluster().stopRandomNode(InternalTestCluster.nameFilter(shrinkNode));
-        ensureStableCluster(nodeCount -1);
+        ensureStableCluster(nodeCount - 1);
 
         // demonstrate that the index.routing.allocation.initial_recovery setting from the shrink doesn't carry over into the split index,
         // because this would cause the shrink to fail as the initial_recovery node is no longer present.

